weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -231,6 +235,16 @@ def kafka_producer(self) -> KafkaProducer:
         self._kafka_producer = KafkaProducer.from_env()
         return self._kafka_producer
 
+    def _noop_project_version_latency_test(self, project_id: str) -> None:
+        # NOOP for testing latency impact of project switcher
+        try:
+            init_resolver(ch_client_factory=lambda: self.ch_client)
```

Comment:
this global singleton is a weird pattern. why not just pass self.ch_client to your `resolve_read_table` method?

Reply:
well two options:

1. just have the router be a member of the ClickhouseTraceServer (this is a singleton in the server process)
2. OR, if you really wanted to force it, just put the `self._cache: LRUCache[str, ProjectDataResidence] = LRUCache(maxsize=cache_size)` in the global space